Matrei/rename proposal

matrei opened the pull request:
Following my discussion with @jdaugherty, I've created a new branch with my renaming ideas.

jdaugherty committed:
artifact rename proposal

matrei committed:
Another suggestion

jdaugherty commented on RENAME.md:
```diff
@@ -0,0 +1,136 @@
+| org.grails                         | tck-base                                        | org.apache.grails          | grails-data-mapping-tck-base                |                        |                          | grails-data-mapping    |
+| org.grails                         | tck-domains                                     | org.apache.grails          | grails-data-mapping-tck-domains             |                        |                          | grails-data-mapping    |
+| org.grails                         | tck-tests                                       | org.apache.grails          | grails-data-mapping-tck-tests               |                        |                          | grails-data-mapping    |
+| org.grails                         | grails-datastore-gorm-hibernate6                | org.apache.grails          | grails-datastore-gorm-hibernate6            |                        |                          | grails-data-hibernate6 |
```
I think in your example, new artifactId should be 'grails-data-mapping-hibernate6'?

jdaugherty commented on RENAME.md:
```diff
@@ -0,0 +1,136 @@
+| org.grails.plugins                 | hibernate6                                      | org.apache.grails          | grails-data-hibernate6                      |                        |                          | grails-data-hibernate6 |
+| org.grails.plugins                 | cache                                           | org.apache.grails          | grails-cache                                |                        |                          | grails-cache           |
+| org.grails.plugins                 | geb                                             | org.apache.grails          | grails-geb                                  |                        |                          | grails-geb             |
+| org.grails.plugins                 | spring-security-acl                             | org.apache.grails          | grails-spring-security-acl                  |                        |                          | grails-spring-security |
```
Should we drop "spring" from the security names?  It seems redundant since we only run on SpringBoot now.

matrei replied:
Yes, we could drop it, but as there are other security frameworks (for example: https://shiro.apache.org/), I thought it could be beneficial to keep it, to make clear what this is based on. I'm not sure how you mean with "since we only run SpringBoot now"?

jamesfredley replied:
I lean slightly towards `spring-security-` for clarity.

jdaugherty replied:
I'd vote keep the spring security then.  Thank you for listening.

jamesfredley committed:
additional changes to grails plugins, CLIs and one mongodb artifactid

jamesfredley commented:
https://github.com/apache/grails-core/pull/14077 some additional changes on top of these

jamesfredley committed:
move view gradle plugins down to grails-gradle-plugin

jamesfredley committed:
move grails-test to org.apache.grails.testing

jamesfredley committed:
cleanup

jamesfredley committed:
more cleanup

jamesfredley committed:
more cleanup

jamesfredley committed:
more cleanup

matrei committed:
Another iteration

- Sort by artifact id (found some duplicates)
- Make sections for GRADLE, FORGE and PROFILES
- Some renames of artifactIds
- Use `json` instead of `gson` as that is the common name
  (gson is the "technology")
- Simplify profile artifactIds as they are not distributed
- Use `gradle-views-json` and `gradle-views-markup` in Gradle
  plugin names.

matrei committed:
Sort views and spring-security into the main table

matrei commented:
Replaced by https://github.com/apache/grails-core/pull/14080
